### ISSUE TITLE
Do multiagent envs support partial step() returns?

### ISSUE THREAD

**Issue by yannbouteiller** [ray-project/ray#6868]
Hello, I am using rllib for its support of multi-agent environments.

I am dealing with a use-case where the step() function will be returning transitions for only a random fraction of the agents present in the environment each time. I have the feeling that this might already work seamlessly with the way rllib samples in multi-agent environments, but I am not sure. Could you confirm or infirm, please?

Thanks!
Yann.

**Comment by ericl** [comment]
Yes, this should work -- you can just return a subset of the agent ids in the dict returned from step. For example, {"agent1": obs} instead of {"agent1": obs, "agent2": obs} if only agent1 is ready.

**Comment by hainesmichaelc** [comment]
Thanks for creating this ticket! In the future, please post questions like this on our [Discourse forum](https://discuss.ray.io/) and someone from the community can help support.
